### PR TITLE
FEATURE: Add displaying CTA

### PR DESCRIPTION
**Pull request by Dawid-Sroka**
https://github.com/Problematy/goodmap-frontend/issues/60

To test how this works, in the database, in `"visible_data"`, add: `"CTA"`.
Then you can add in any datapoint `"data"` section this attribute:
```
                "CTA": {
                    "type": "CTA",
                    "value": "<address-to-page>",
                    "displayValue": "<Call to action>"
                }
```

### CODE REVIEW

File: src/components/MarkerPopup/mapCustomTypeToReactComponent.jsx [hunk 1]
```diff
@@ -16,6 +17,16 @@ export const mapCustomTypeToReactComponent = customValue => {
...
                 </a>
             );
+        case 'CTA':
+            return (
+                <button
+                    onClick={() => alert('Button clicked!')}
+                    style={MarkerCTAButtonStyle}
+                    variant="contained"
+                >
+                    CTA
```
Comment (raven-wing) [inline src/components/MarkerPopup/mapCustomTypeToReactComponent.jsx:27]: There should be "display_value" probably shown instead CTA

**Review by raven-wing — COMMENTED** [review]
Add tests which are clicking this button. and are checking that it redirects to proper page

File: src/components/MarkerPopup/mapCustomTypeToReactComponent.jsx (hunk 1 above)
Comment (raven-wing) [reply src/components/MarkerPopup/mapCustomTypeToReactComponent.jsx:27]: Also - this button should redirect somewhere for now (later there will be form popup, but for now redirection should be enough).

**Review by raven-wing — COMMENTED** [review]
Add tests which are clicking this button. and are checking that it redirects to proper page

File: tests/CTA.test.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    data: [
+        ['length', 112.5],
+        ['accessible_by', ['pedestrians', 'cars']],
+        ['website', { type: 'hyperlink', value: 'https://www.google.com' }],
+        [
+            'CTA',
+            {
+                type: 'CTA',
+                value: 'https://www.example.com',
+                displayValue: 'Oddaj przez example.org!',
```
Comment (raven-wing) [inline tests/CTA.test.jsx:19]:
```suggestion
                displayValue: 'Visit example.org!',
```

**Comment by raven-wing** [comment]
2024-10-23-151007_255x53_scrot

That's how it looks now.

For sure it should not have "CTA:" on the left. This is special button that is added to some pins.

As for functionality - it probably should open new tab instad working in same.

**Comment by raven-wing** [comment]
Ok. Now it looks like that for me:
2024-10-25-081639_198x126_scrot


Which is exact opposite what we wanted to acheive. Button was good. CTA label was bad :)

**Comment by Dawid-Sroka** [comment]
> Ok. Now it looks like that for me:

Please show me the data json for 'Zwierzyniecka' you are testing.

**Comment by raven-wing** [comment]
```
            {
                "name": "Zwierzyniecka",
                "position": [
                    51.10655,
                    17.0555
                ],
                "accessible_by": [
                    "bikes",
                    "pedestrians"
                ],
                "type_of_place": "small bridge",
                "CTA": {
                    "type": "CTA",
                    "value": "https://www.example.com",
                    "displayValue": "Visit example.org!"
                },
                "UUID": "dattarro"
            }

```

**Comment by raven-wing** [comment]
2024-10-25-100914_348x310_scrot

**Comment by raven-wing** [comment]
It works. My bad

File: tests/CTA.test.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    },
+};
+
+describe('CTA', () => {
+    it('should redirect to a page specified by CTA when CTA button clicked', async () => {
+        const mockOpen = jest.spyOn(window, 'open').mockImplementation(() => {});
+
+        const { getByText } = render(<MarkerContent place={correctMarkerData} />);
+
+        const button = getByText('Oddaj przez example.org!');
```
Comment (raven-wing) [inline tests/CTA.test.jsx:34]:
```suggestion
        const button = getByText('Visit example.org!');
```

**Review by raven-wing — APPROVED** [review]

**Comment by raven-wing** [comment]
It closed https://github.com/Problematy/goodmap/issues/156 (it was not needed)

**Pull request merged**